### PR TITLE
Add dynamic viscosity quantity with related units and tests

### PR DESCRIPTION
**Pull request by crystal-growth**
Dynamic viscosity quantity [Pressure x Time] with derived and imperial units.

### CODE REVIEW

**Comment by crystal-growth** [comment]
Rebased to current master, fixing merge conflict in mod.rs.

File: src/si/dynamic_viscosity.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
+//! Dynamic viscosity (base unit pascal second, kg · m⁻¹ · s⁻¹).
+
+quantity! {
+    /// Dynamic viscosity (base unit pascal second, kg · m⁻¹ · s⁻¹).
+    quantity: DynamicViscosity; "dynamic viscosity";
+    /// Dimension of dynamic viscosity, ML⁻¹T⁻¹ (base unit pascal second, kg · m⁻¹ · s⁻¹).
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:6]: Dimensions in ISQ order.

```suggestion
    /// Dimension of dynamic viscosity, L⁻¹MT⁻¹ (base unit pascal second, kg · m⁻¹ · s⁻¹).
```

File: src/si/dynamic_viscosity.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        @reyn: 4.448_222_E0 / 2.54_E-2 / 2.54_E-2; "reyn",
+            "reyn", "reyn";
+        @pound_per_foot_second: 4.535_924_E-1 / 3.048_E-1; "lb/(ft·s)",
+            "pound per foot second", "pounds per foot second";      
+        @pound_per_foot_hour: 4.535_924_E-1 / 3.048_E-1 / 3600.0; "lb/(ft·h)",
+            "pound per foot hour", "pounds per foot hour";
+        @slug_per_foot_second: 1.459_390_E1 / 3.048_E-1; "slug/(ft·s)",
+            "slug per foot second", "slugs per foot second";  
+        @gram_per_centimeter_second: prefix!(milli) / prefix!(centi); "g/(cm·s)",
+            "gram per centimeter second", "grams per centimeter second";
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:42]: * Formatting
 * In my brief searching I couldn't find any reference about `poises` being the plural of `poise` and `reyn` of `reyn`. Will you verify?

```suggestion
        @pascal_second: prefix!(none); "Pa · s", "pascal second", "pascal seconds";
        @millipascal_second: prefix!(milli); "mPa · s", "millipascal second", "millipascal seconds";
        @micropascal_second: prefix!(micro); "µPa · s", "micropascal second", "micropascal seconds";
        // poise = 0.1 Pa · s
        @poise: 1.0_E-1; "P", "poise", "poises";
        // centipoise = 1 mPa · s
        @centipoise: prefix!(centi) * 1.0_E-1; "cP", "centipoise", "centipoises";
        @pound_force_second_per_square_foot: 4.448_222_E0 / 3.048_E-1 / 3.048_E-1; "lbf · s/ft²",
            "pound-force second per square foot", "pound-force seconds per square foot";
        @pound_force_second_per_square_inch: 4.448_222_E0 / 2.54_E-2 / 2.54_E-2; "lbf · s/in²",
            "pound-force second per square inch", "pound-force seconds per square inch";
        // Reyn = 1 lbf · s/in²
        @reyn: 4.448_222_E0 / 2.54_E-2 / 2.54_E-2; "reyn", "reyn", "reyn";
        @pound_per_foot_second: 4.535_924_E-1 / 3.048_E-1; "lb/(ft · s)", "pound per foot second",
            "pounds per foot second";
        @pound_per_foot_hour: 4.535_924_E-1 / 3.048_E-1 / 3.6_E3; "lb/(ft · h)",
            "pound per foot hour", "pounds per foot hour";
        @slug_per_foot_second: 1.459_390_E1 / 3.048_E-1; "slug/(ft · s)", "slug per foot second",
            "slugs per foot second";
        @gram_per_centimeter_second: prefix!(milli) / prefix!(centi); "g/(cm · s)",
            "gram per centimeter second", "grams per centimeter second";
```

File: src/si/dynamic_viscosity.rs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        #[test]
+        fn check_dimension() {
+            let _: DynamicViscosity<V> = Pressure::new::<p::pascal>(V::one())
+                * Time::new::<t::second>(V::one());
+        }
+
+        #[test]
+        fn check_units() {
+            test::<dv::pascal_second, t::second, p::pascal>();
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:66]: I'd like to see the generic parameters in formula order: pressure * time = dynamic viscosity.

```suggestion
            test::<p::pascal, t::second, dv::pascal_second>();
```

File: src/si/dynamic_viscosity.rs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            test::<dv::millipascal_second, t::second, p::millipascal>();
+            test::<dv::micropascal_second, t::second, p::micropascal>();
+            test::<dv::poise, t::second, p::dyne_per_square_centimeter>();
+            test::<dv::centipoise, t::second, p::millipascal>();
+            test::<dv::pound_force_second_per_square_foot, t::second, p::pound_force_per_square_foot>();
+            test::<dv::pound_force_second_per_square_inch, t::second, p::pound_force_per_square_inch>();
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:72]: Wrap long lines.

```suggestion
            test::<dv::pound_force_second_per_square_foot, t::second,
                p::pound_force_per_square_foot>();
            test::<dv::pound_force_second_per_square_inch, t::second,
                p::pound_force_per_square_inch>();
```

File: src/si/dynamic_viscosity.rs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            test::<dv::slug_per_foot_second, t::second, m::slug, l::foot>();
+
+            fn test<DV: dv::Conversion<V>, T: t::Conversion<V>, M: m::Conversion<V>, L: l::Conversion<V>>() {
+                Test::assert_approx_eq(&DynamicViscosity::new::<DV>(V::one()),
+                    &(Mass::new::<M>(V::one())
+                        / Length::new::<L>(V::one())
+                        / Time::new::<T>(V::one())));
+            }
+        }        
+
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:96]:
```suggestion
        }
```

File: src/si/dynamic_viscosity.rs [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            fn test<DV: dv::Conversion<V>, T: t::Conversion<V>, P: p::Conversion<V>>() {
+                Test::assert_approx_eq(&DynamicViscosity::new::<DV>(V::one()),
+                    &(Pressure::new::<P>(V::one())
+                        * Time::new::<T>(V::one())));
+            }
+        }
+
+        #[test]
+        fn check_units_mlt() {
+            test::<dv::pound_per_foot_second, t::second, m::pound, l::foot>();
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:84]: Generic parameters in formula order again.

```suggestion
            test::<m::pound, l::foot, t::second, dv::pound_per_foot_second>();
```

File: src/si/dynamic_viscosity.rs [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            test::<dv::pound_per_foot_hour, t::hour, m::pound, l::foot>();
+            test::<dv::gram_per_centimeter_second, t::second, m::gram, l::centimeter>();
+            test::<dv::slug_per_foot_second, t::second, m::slug, l::foot>();
+
+            fn test<DV: dv::Conversion<V>, T: t::Conversion<V>, M: m::Conversion<V>, L: l::Conversion<V>>() {
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:89]: Wrap long line.

```suggestion
            fn test<DV: dv::Conversion<V>, T: t::Conversion<V>, M: m::Conversion<V>,
                L: l::Conversion<V>>()
            {
```

or

```suggestion
            fn test<
                DV: dv::Conversion<V>,
                T: t::Conversion<V>,
                M: m::Conversion<V>,
                L: l::Conversion<V>>()
            {
```

**Review by iliekturtles — CHANGES_REQUESTED** [review]

**Comment by crystal-growth** [comment]
I'm not sure about plurals for Poise
"Poises" is frequently used, especially in older literature:
https://ceramics.onlinelibrary.wiley.com/doi/abs/10.1111/j.1151-2916.1963.tb11684.x
https://www.sciencedirect.com/science/article/pii/001910359190090G
Same for reyn: 
"Reyns" are used in older texts: https://ntrs.nasa.gov/api/citations/19930083501/downloads/19930083501.pdf

The newer texts use standard SI units or abbreviations

File: src/si/dynamic_viscosity.rs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        fn check_dimension() {
+            let _: DynamicViscosity<V> = Pressure::new::<p::pascal>(V::one())
+                * Time::new::<t::second>(V::one());
+        }
+
+        #[test]
+        fn check_units() {
+            test::<p::pascal, t::second, dv::pascal_second>();
+            test::<p::millipascal, t::second, dv::millipascal_second>();
+            test::<p::micropascal,  t::second, dv::micropascal_second>();
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:62]:
```suggestion
            test::<p::micropascal, t::second, dv::micropascal_second>();
```

File: src/si/dynamic_viscosity.rs [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+            test::<p::dyne_per_square_centimeter, t::second, dv::poise>();
+            test::<p::millipascal, t::second,dv::centipoise>();
+            test::<p::pound_force_per_square_foot, t::second,
+                dv::pound_force_second_per_square_foot>();
+            test::<p::pound_force_per_square_inch,  t::second,
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:67]:
```suggestion
            test::<p::pound_force_per_square_inch, t::second,
```

File: src/si/dynamic_viscosity.rs [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+                        * Time::new::<T>(V::one())));
+            }
+        }
+
+        #[test]
+        fn check_units_mlt() {
+            test::<m::pound, l::foot, t::second, dv::pound_per_foot_second>();
+            test::<m::pound, l::foot, t::hour,  dv::pound_per_foot_hour>();
+            test::< m::gram, l::centimeter,  t::second, dv::gram_per_centimeter_second>();
+            test::< m::slug, l::foot, t::second, dv::slug_per_foot_second>();
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:83]:
```suggestion
            test::<m::pound, l::foot, t::hour, dv::pound_per_foot_hour>();
            test::<m::gram, l::centimeter, t::second, dv::gram_per_centimeter_second>();
            test::<m::slug, l::foot, t::second, dv::slug_per_foot_second>();
```

File: src/si/dynamic_viscosity.rs [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+            fn test<M: m::Conversion<V>, L: l::Conversion<V>, T: t::Conversion<V>, DV: dv::Conversion<V>>()
```
Comment (iliekturtles) [inline src/si/dynamic_viscosity.rs:85]:
```suggestion
            fn test<M: m::Conversion<V>, L: l::Conversion<V>, T: t::Conversion<V>,
                DV: dv::Conversion<V>>()
```

**Review by iliekturtles — CHANGES_REQUESTED** [review]
Just some minor formatting then I think we're ready to merge!

**Comment by crystal-growth** [comment]
Thank you! 
Rebased and squashed into one commit.

**Comment by iliekturtles** [comment]
Thanks so much for this PR!

**Pull request merged**